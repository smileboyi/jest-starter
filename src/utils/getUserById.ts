const getUserById = async (id: string) => {
  return Promise.resolve({
    params: { id },
  });
};

export default getUserById;
