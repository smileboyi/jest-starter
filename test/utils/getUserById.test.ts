import getUserById from "utils/getUserById";

describe("getUserById", () => {
  it("可以获取 userId == 1 的用户", async () => {
    const result = await getUserById("1");
    // 纯文本、JSON、XML等能被序列化的内容都可以使用快照测试
    expect(result).toMatchSnapshot();
  });
});
