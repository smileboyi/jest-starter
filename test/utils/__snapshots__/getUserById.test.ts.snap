// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUserById 可以获取 userId == 1 的用户 1`] = `
Object {
  "params": Object {
    "id": "1",
  },
}
`;
